File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/RecordingImpl.kt
package dev.hebei.camerax_android.video

import androidx.camera.video.Recording
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiRecordingApi

class RecordingImpl(impl: CameraXImpl) : PigeonApiRecordingApi(impl) {
    @androidx.camera.video.ExperimentalPersistentRecording
    override fun isPersistent(pigeon_instance: Recording): Boolean {
        return pigeon_instance.isPersistent
    }

    override fun mute(pigeon_instance: Recording, muted: Boolean) {
        return pigeon_instance.mute(muted)
    }

    override fun pause(pigeon_instance: Recording) {
        return pigeon_instance.pause()
    }

    override fun resume(pigeon_instance: Recording) {
        return pigeon_instance.resume()
    }

    override fun stop(pigeon_instance: Recording) {
        return pigeon_instance.stop()
    }
}

File: camerax_ios/lib/src/video/recording_impl.dart
import 'package:camerax_ios/src/api.dart';
import 'package:camerax_ios/src/common.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class RecordingImpl extends AutoCloseableImpl implements Recording {
  @override
  final RecordingProxyApi api;

  RecordingImpl.internal(this.api);

  @override
  Future<bool> isPersistent() => api.isPersistent();

  @override
  Future<void> mute(bool muted) => api.mute(muted);

  @override
  Future<void> pause() => api.pause();

  @override
  Future<void> resume() => api.resume();

  @override
  Future<void> stop() => api.stop();
}

extension RecordingProxyApiX on RecordingProxyApi {
  RecordingImpl get impl => RecordingImpl.internal(this);
}
